File: paystack/src/main/java/co/paystack/android/api/service/ApiService.java
package co.paystack.android.api.service;

import java.util.HashMap;

import co.paystack.android.api.model.TransactionApiResponse;
import retrofit2.Call;
import retrofit2.http.FieldMap;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;

/**
 * ApiService
 */
public interface ApiService {

    @FormUrlEncoded
    @POST("/charge/mobile_charge")
    Call<TransactionApiResponse> charge(@FieldMap HashMap<String, String> fields);

    @FormUrlEncoded
    @POST("/charge/validate")
    Call<TransactionApiResponse> validateCharge(@FieldMap HashMap<String, String> fields);

    @GET("/requery/{trans}")
    Call<TransactionApiResponse> requeryTransaction(@Path("trans") String trans);

    @FormUrlEncoded
    @POST("/charge/avs")
    Call<TransactionApiResponse> submitCardAddress(@FieldMap HashMap<String, String> fields);

}

File: paystack/src/main/java/co/paystack/android/api/service/ApiService.kt
package co.paystack.android.api.service

import co.paystack.android.api.converter.Wrapped
import co.paystack.android.api.model.TransactionApiResponse
import co.paystack.android.api.model.TransactionInitResponse
import retrofit2.Call
import retrofit2.http.*
import java.util.*

/**
 * ApiService
 */
interface ApiService {
    @FormUrlEncoded
    @POST("/charge/mobile_charge")
    fun charge(@FieldMap fields: HashMap<String, String>): Call<TransactionApiResponse>

    @FormUrlEncoded
    @POST("/charge/validate")
    fun validateCharge(@FieldMap fields: HashMap<String, String>): Call<TransactionApiResponse>

    @GET("/requery/{trans}")
    fun requeryTransaction(@Path("trans") trans: String?): Call<TransactionApiResponse>

    @Wrapped
    @GET("/checkout/request_inline")
    suspend fun initiateTransaction(@QueryMap params: Map<String, @JvmSuppressWildcards Any>): TransactionInitResponse
}